### PR TITLE
reports: Add li-boxuan's GSoC report

### PR DESCRIPTION
**Pull request by li-boxuan**
https://deploy-preview-618--coala-projects.netlify.com/gsoc/2018/statusreport/li-boxuan

### CODE REVIEW

**Comment by li-boxuan** [comment]
Preview: https://deploy-preview-618--coala-projects.netlify.com/gsoc/2018/statusreport/li-boxuan

File: reports/li-boxuan.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
+---
+layout: report
+permalink: /GSoC/2018/StatusReport/li-boxuan
```
Comment (hemangsk) [inline reports/li-boxuan.md]: I think you have to add a report_year variable as well

**Comment by jayvdb** [comment]
You could include commits in https://gitlab.com/coala/coala-json/commits/master and  https://github.com/coala/coala-vs-code/commit/59660ef29f169459b4c3690503eea47a07272cf9 and https://github.com/coala/coala-vs-code/commit/f90e5a2cdcdeda3340955b8f0a7bdf7e10d3d863 which were related to your offset work.

IIRC you also raised an issue about pylint -- IMO that is also worth noting.  You could add a paragraph about it as auxillary work done as part of the community, and it was very important work to improve quality of coala output.

**Comment by jayvdb** [comment]
Also you sent some patches upstream.  They should be noted, even though they havent been merged.

File: reports/li-boxuan.md [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+   - username: li-boxuan
+   - link: https://gitter.im/li-boxuan
+ - Openhub:
+   - username: li-boxuan
+   - link: https://www.openhub.net/accounts/li-boxuan
+email: liboxuanhk@gmail.com
+blog: http://li-boxuan.github.io/
+activity:
+
+ - 0:
```
Comment (jayvdb) [inline reports/li-boxuan.md]: start at 1

also this is yaml, which has two space indents

File: reports/li-boxuan.md [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+    - details: >
+       CheckStyleBear: Fix offset convention
+
+  - 48:
+    - repo: philschatz/gh-board (upstream)
+    - link: https://github.com/philschatz/gh-board/pull/147
+    - details: >
+       Modernise React code, update packages, fix warnings and improve build (open)
+
+  - 49:
```
[the remark this answers is not in the document]
Comment (li-boxuan) [reply reports/li-boxuan.md]: 1. > Some ... are too small to mention?

Yes, I have several small commits which just do gitignore sync, so I didn't mention them in the report.

2.  > are these upstream PRs in your tarball

I didn't put upstream PRs in my tarball. 

3. > I suggest you move these two upstream PRs into the prose below

Sounds good. Will do that.

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**